File: update_menus.py
from os import walk
import subprocess

def write(fp, data):
    f = open(fp, "w", encoding="utf-8")
    f.write(data)
    f.close()

def main():
    
    dirs = next(walk("restaurants"))[1]

    menus = "["

    for directory in dirs:
        print(directory)
        completed = subprocess.run(
            ["python", "restaurants/" + directory + "/get_menu.py"],
            capture_output=True,
            text=True)
        text = completed.stdout
        menus += text + ","
    menus = menus[:-1] + "]"

    menus = "".join(menus.split("\n"))

    write("public/restaurants.json", menus)

main()

File: updateMenus.js

const fs = require("fs")
const glob = require("glob")


function update() {
    glob("restaurants/*", (err, dirs) => {

        let menus = []

        dirs.forEach(dir => {
            let menuGetter = require("./" + dir + "/getMenu.js")
            menus.push(menuGetter())
        })

        Promise.all(menus).then(values => {
            fs.writeFile("./public/menus.json", JSON.stringify(values), () => {})
        })
    })
}


update()

setInterval(update, 3600000)

